app/app.js

```diff
@@ -61,8 +61,20 @@ function load(stats) {
 			if(!m) return;
 			origin.moduleUid = m.uid;
 		});
+		chunk.modules.forEach(function(module) {
+			var m = mapModulesIdent["$"+module.identifier], s;
+			if(!m) return;
+			s = m.dependencies.reduce(function(totalSize, dep) {
```

Comment:
So that shouldn't be a problem since we're looking at modules and the chunks they appear in, instead of chunks and their parents.

One potential issue though is that 'recursive' is kind of a misnomer since we're only traversing one step down in the dependency tree here. I mistakenly assumed that webpack was including the recursive dependencies in the stats output, but it looks like only the immediate ones. We can travel further down the tree but I'll have to keep track so that a module is only included once in a size calculation and to avoid circular calculations. I think that will be my next step with this, since otherwise the recursive size won't be 'normalized' to include jquery and such vendor libs that are likely to be included in pretty much all modules after recursive analysis of the size.